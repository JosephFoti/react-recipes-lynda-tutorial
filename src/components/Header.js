import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logo from '../static/images/dog.jpg';

const HeaderLink = ({ children, ...props }) => (
  <NavLink
    exact
    className="p1 mx2 black rounded text-decoration-none"
    activeClassName="bg-white"
    {...props}
  >
    {children}
  </NavLink>
);

const Header = () => (
  <header className="flex items-center justify-between px4">
    <div>
      <h1 className="h1 inline-block mr2">🍽 My Recipes</h1>
      <span className="mr2">by</span>
      <img src={Logo} style={{ width: 30 }} alt="A dog dressed like a man" />
    </div>
    <nav>
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/favorites">Favorites</HeaderLink>
    </nav>
  </header>
);

HeaderLink.propTypes = {
  children: PropTypes.node,
};

export default Header;
