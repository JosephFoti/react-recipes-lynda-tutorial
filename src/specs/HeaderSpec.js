import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('<Header />', () => {
  test('Should render correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
