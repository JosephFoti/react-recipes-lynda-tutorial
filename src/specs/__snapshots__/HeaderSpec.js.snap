// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render correctly 1`] = `
<header
  className="flex items-center justify-between px4"
>
  <div>
    <h1
      className="h1 inline-block mr2"
    >
      üçΩ My Recipes
    </h1>
    <span
      className="mr2"
    >
      by
    </span>
    <img
      alt="A dog dressed like a man"
      src="test-file-stub"
      style={
        Object {
          "width": 30,
        }
      }
    />
  </div>
  <nav>
    <a
      aria-current={null}
      className="p1 mx2 black rounded text-decoration-none"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="p1 mx2 black rounded text-decoration-none"
      href="/favorites"
      onClick={[Function]}
    >
      Favorites
    </a>
  </nav>
</header>
`;
